packages/environment/test/assets/price-feed.test.ts

```diff
@@ -121,11 +124,38 @@ suite.each(assets)("$symbol ($name): $id", (asset) => {
         break;
       }
 
-      case PriceFeedType.DERIVATIVE_BALANCER_V2_STABLE_POOL:
+      case PriceFeedType.DERIVATIVE_BALANCER_V2_STABLE_POOL: {
+        const [priceFeed, poolInfo] = await Promise.all([
+          Protocol.getPriceFeedForDerivative(client, { valueInterpreter, asset: asset.id }),
+          BalancerV2.getPoolInfo(client, { balancerV2StablePoolPriceFeed: asset.priceFeed.address, pool: asset.id }),
+        ]);
+
+        expect(toAddress(priceFeed)).toBe(asset.priceFeed.address);
+        expect(toAddress(poolInfo.invariantProxyAsset)).toBe(asset.priceFeed.ipa);
+
+        const isSupportedAssetForDerivativePriceFeed = await Protocol.isSupportedAssetForDerivativePriceFeed(client, {
+          derivativePriceFeed: priceFeed,
+          asset: asset.id,
+        });
+
+        expect(isSupportedAssetForDerivativePriceFeed).toBe(true);
+        const ipa = asset.priceFeed.ipa;
+        // Check that the invariant proxy asset exists in the asset universe, or that it is equal to the UsdEthSimulatedAggregator
+        // (but not both)
+        expect(ipa === usdEthSimulatedAggregator || environment.hasAsset(ipa)).toBe(true);
+
+        break;
+      }
+
       case PriceFeedType.DERIVATIVE_CURVE: {
-        const priceFeed = await Protocol.getPriceFeedForDerivative(client, { valueInterpreter, asset: asset.id });
+        const pool = (asset as CurvePoolLpAsset).pool;
```

Comment:
`const pool = environment.getAssetAs(asset.id, AssetType.CURVE_POOL_LP).pool;` maybe we could do it like this, so the error will be thrown if it won't curve lp asset

Reply:
Good idea, but we also need to take into account that the asset could also be a curve gauge.

Reply:
Fixed this by using another test: `expect([AssetType.CURVE_POOL_LP, AssetType.CURVE_POOL_GAUGE].includes(asset.type)).toBe(true);`